lib/compat/wordpress-6.2/class-gutenberg-rest-global-styles-controller-6-2.php

```diff
@@ -1,304 +0,0 @@
-<?php
-/**
- * REST API: Gutenberg_REST_Global_Styles_Controller class
- *
- * @package    Gutenberg
- * @subpackage REST_API
- */
-
-/**
- * Base Global Styles REST API Controller.
- */
-class Gutenberg_REST_Global_Styles_Controller_6_2 extends WP_REST_Global_Styles_Controller {
```

Comment:
Instead of removing it, we need to make the global styles' endpoint evergreen, like `global-styles-and-settings.php` is. I thought it was already the case.

This is why: as long as the underlying data and how it is processed changes (`theme.json`, `WP_Theme_JSON_Gutenberg`, and `WP_Theme_JSON_Resolver_Gutenberg`), everything that sits on top of it needs to use the Gutenberg classes.

Example: imagine we add a new key `myNewKey` for `theme.json` in Gutenberg. If the REST API endpoint is not using the Gutenberg classes (schema, sanitizer, etc.) that key will be removed by the core classes. Same if we change how styles are generated, for example.

Sorry that I can pinpoint specifics of how this would break, I have a limited view of what changed as I wasn't involved in this area for the past few cycles. Hoping this helps anyway?

Reply:
Ah yes, good point, thanks. I'll fix that up.

Reply:
I've added a PR here: https://github.com/WordPress/gutenberg/pull/53618